src/main/java/ru/skypro/homework/controller/CommentsController.java

```diff
@@ -32,55 +32,66 @@ public class CommentsController {
+            return ResponseEntity.ok(commentsDTO);
+        } catch (RuntimeException e) {
+            e.getStackTrace();
+            return ResponseEntity.status(HttpStatus.FORBIDDEN).build();
+        }
+
     }
 
     // добавление комментариев
     @Operation(summary = "Добавление нового комментария")
     @PostMapping("/{id}/comments")
-    public ResponseEntity<?> addComment(@PathVariable Comments pk, @RequestBody String text) {
-        if (pk != null) {
-            return ResponseEntity.ok(pk);
-        } else return ResponseEntity.status(HttpStatus.NOT_FOUND).build();
+    public ResponseEntity<CommentDTO> addComment(@PathVariable Integer pk,
+                                                 @RequestBody CreateOrUpdateCommentDTO COUComment,
+                                                 String userInfo ) {
+        try {
+            CommentDTO commentDTO = commentsService.addComment(COUComment, pk, userInfo);
+            return ResponseEntity.ok(commentDTO);
+        } catch (RuntimeException e) {
+            e.getStackTrace();
+            return ResponseEntity.status(HttpStatus.FORBIDDEN).build();
+        }
     }
 
     // удаление комментария по id
     @Operation(summary = "Удаление комментария")
     @DeleteMapping("/{adId}/comments/{commentId}")
 //    public void deleteComment(@PathVariable int adId , @RequestParam int commentId ) {
-    public ResponseEntity<?> deleteComment(@PathVariable int adId , @PathVariable int pk ){
-        adsService.getAdById(adId);
-        boolean deleteIsOk = commentsService.deleteComment(pk);
-        if (deleteIsOk) {
-            return new ResponseEntity<>(("Комментарий с id = " + pk + "успешно удалён"), HttpStatus.NO_CONTENT);
-        } else {
-            return new ResponseEntity<>(("Ошибка при попытке удалить комментарий с id = " + pk), HttpStatus.NOT_FOUND);
+    public ResponseEntity<CommentDTO> deleteComment(@PathVariable int adId , @PathVariable int pk ) {
+        try {
+            commentsService.deleteComment(adId, pk);
+            return ResponseEntity.ok().build();
+        } catch (RuntimeException e) {
+            e.getStackTrace();
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).build();
         }
     }
 
     // обновление комментария
-    @Operation(summary = "Обновление комментария")
-    @PatchMapping("/{adId}/comments/{commentId}")
-    public ResponseEntity<?> updateComment(@PathVariable int adId,@PathVariable int pk, @RequestBody String text) {
-        adsService.getAdById(adId);
-        commentsService.getComment(pk);
-        CreateOrUpdateCommentDTO updateComment = new CreateOrUpdateCommentDTO(text);
-        return ResponseEntity.ok(commentsService.updateComment(pk, updateComment));
+        @Operation(summary = "Обновление комментария")
+        @PatchMapping("/{adId}/comments/{commentId}")
+    public ResponseEntity<CommentDTO> updateComment(@PathVariable int adId,@PathVariable int pk,
```

Comment:
pk - неправильное название параметра, такого параметра нет в пути

Reply:
Понял, когда писал, думал поменять на commentId , но решил так оставить, получается всё же исправить надо было?

Reply:
маппинг происходит либо по названию переменной, либо через аннотацию. Т.е.
```suggestion
    public ResponseEntity<CommentDTO> updateComment(@PathVariable int adId, @PathVariable int commentId,
```
или
```suggestion
    public ResponseEntity<CommentDTO> updateComment(@PathVariable int adId, @PathVariable("commentId") int pk,
```